app/controlplane/pkg/biz/attestation.go

```diff
@@ -44,16 +44,36 @@ func NewAttestationUseCase(client CASClient, logger log.Logger) *AttestationUseC
 	}
 }
 
-func (uc *AttestationUseCase) UploadToCAS(ctx context.Context, envelope *dsse.Envelope, backend *CASBackend, workflowRunID string) (*cr_v1.Hash, error) {
-	filename := fmt.Sprintf("attestation-%s.json", workflowRunID)
+func (uc *AttestationUseCase) UploadEnvelopeToCAS(ctx context.Context, envelope *dsse.Envelope, backend *CASBackend, workflowRunID string) (string, error) {
 	jsonContent, h, err := attestation.JSONEnvelopeWithDigest(envelope)
 	if err != nil {
-		return nil, fmt.Errorf("marshaling the envelope: %w", err)
+		return "", fmt.Errorf("marshaling the envelope: %w", err)
 	}
 
-	if err := uc.CASClient.Upload(ctx, string(backend.Provider), backend.SecretName, bytes.NewBuffer(jsonContent), filename, h.String()); err != nil {
-		return nil, fmt.Errorf("uploading to CAS: %w", err)
+	if err = uc.doUploadToCAS(ctx, fmt.Sprintf("attestation-%s.json", workflowRunID), jsonContent, backend, h.String()); err != nil {
+		return "", fmt.Errorf("uploading to CAS: %w", err)
 	}
 
-	return &h, nil
+	return h.String(), nil
+}
+
+func (uc *AttestationUseCase) UploadBundleToCAS(ctx context.Context, bundle *protobundle.Bundle, backend *CASBackend, workflowRunID string) (string, error) {
+	jsonContent, h, err := attestation.JSONBundleWithDigest(bundle)
```

Comment:
as I said, it's ok but literally this method does not need to receive a bundle, nor a envelope in the other case, a proto.Message would have worked.

Reply:
Note that `dsse.Envelope` is NOT a proto message. That's why we need two different paths. However, we could extract the json content outside and have one single `UploadToCAS`

Reply:
true, also we'll remove one of those methods soon, it makes sense